pkg/shp/cmd/build/run.go

```diff
@@ -71,6 +81,49 @@ func (c *RunCommand) Run(params *params.Params, ioStreams *genericclioptions.IOS
 		return err
 	}
 
-	fmt.Fprintf(ioStreams.Out, "BuildRun created %q\n", buildRun.Name)
+	if !c.Follow {
+		fmt.Fprintf(ioStreams.Out, "BuildRun created %q\n", buildRun.Name)
+		return nil
+	}
+
+	err := wait.PollImmediate(1*time.Second, 30*time.Second, func() (bool, error) {
+		if err := brr.Get(c.cmd.Context(), buildRun.Name, buildRun); err != nil {
+			fmt.Fprintf(ioStreams.ErrOut, "BuildRun %q not available: %q\n", buildRun.Name, err.Error())
+			return false, nil
+		}
+
+		if buildRun.Status.LatestTaskRunRef == nil {
+			fmt.Fprintf(ioStreams.ErrOut, "BuildRun %q created but cannot follow logs there is no TaskRun reference\n", buildRun.Name)
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		return err
+	}
+
+	tparams := &tkncli.TektonParams{}
+	tparams.SetNamespace(buildRun.Namespace)
+	opts := &tknopt.LogOptions{
+		AllSteps:    false,
+		Follow:      true,
+		Params:      tparams,
+		TaskrunName: *buildRun.Status.LatestTaskRunRef,
+		Stream: &tkncli.Stream{
+			In:  ioStreams.In,
+			Out: ioStreams.Out,
+			Err: ioStreams.ErrOut,
+		},
+	}
+	lr, err := tknlog.NewReader(tknlog.LogTypeTask, opts)
```

Comment:
See https://github.com/tektoncd/cli/blob/3bea8d3c95a83e1b5bdf488e2c5955e897034911/pkg/cmd/pipelinerun/logs.go#L100-L110 for the rest of these changes from here to the end of the method were derived